### ISSUE TITLE
Create Cucumber/Gherkin writeups for first 2 key user workflows/journeys

### ISSUE THREAD

**Issue by glennmatthews** [nautobot/nautobot#4781]
From #23, create the the test cases for these two:

- Build out a location hierarchy
- Build out a prefix/namespace hierarchy

**Comment by glennmatthews** [comment]
First draft:

```gherkin
Feature: Locations
  In order to represent the geographical structure of my organization's infrastructure,
  As Nelly the Network Engineer
  I want to record in Nautobot an appropriate set of LocationType and Location records.

  Background:
    Given Nelly has appropriate Nautobot permissions for a Network Engineer
    And Nelly is logged in to Nautobot

  Scenario: Creating a root LocationType
    When Nelly uses the UI to create a LocationType with name Nation
    Then LocationType Nation should exist in Nautobot

  Scenario: Creating an intermediate LocationType
    Given LocationType Nation exists in Nautobot
    When Nelly uses the UI to create a LocationType with name City
    And Nelly specifies that this LocationType has parent Nation
    Then LocationType City should exist in Nautobot
    And LocationType City should have parent Nation

  Scenario: Creating a leaf LocationType
    Given LocationType Nation exists in Nautobot
    And LocationType City exists in Nautobot with parent Nation
    When Nelly uses the UI to create a LocationType with name Building
    And Nelly specifies that this LocationType has parent City
    And Nelly specifies that Devices can belong to Locations of type Building
    Then LocationType Building should exist in Nautobot
    And LocationType Building should have parent City
    And LocationType Building should show that Devices can belong to Locations of type Building

  Scenario: Creating a root Location
    Given LocationType Nation exists in Nautobot
    When Nelly uses the UI to create a Location with name "United States"
    And Nelly specifies that this Location has location_type Nation
    Then Location United States should exist in Nautobot
    And Location United States should have location_type Nation

  Scenario: Creating an intermediate Location
    Given LocationType Nation exists in Nautobot
    And Location "United States" exists in Nautobot with location_type Nation
    And LocationType City exists in Nautobot with parent Nation
    When Nelly uses the UI to create a Location with name Durham
    And Nelly specifies that this Location has location_type City
    And Nelly specifies that this Location has parent "United States"
    Then Location Durham should exist in Nautobot
    And Location Durham should have location_type City
    And Location Durham should have parent "United States"
    And Location "United States" should show Durham as one of its children

  Scenario: Creating a leaf Location
    Given LocationType Nation exists in Nautobot
    And Location "United States" exists in Nautobot with location_type Nation
    And LocationType City exists in Nautobot with parent Nation
    And Location Durham exists in Nautobot with location_type City and parent Location "United States"
    And LocationType Building exists in Nautobot with parent City
    When Nelly uses the UI to create a Location with name "Durham Bulls Athletic Park"
    And Nelly specifies that this Location has location_type Building
    And Nelly specifies that this Location has parent Durham
    Then Location "Durham Bulls Athletic Park" should exist in Nautobot
    And Location "Durham Bulls Athletic Park" should have location_type Building
    And Location "Durham Bulls Athletic Park" should have parent Durham
    And Location Durham should show "Durham Bulls Athletic Park" as one of its children
```

**Comment by glennmatthews** [comment]
Second draft:

```gherkin
  In order to represent the geographical structure of my organization's infrastructure,
  As a Network Engineer
  I want to record in Nautobot an appropriate set of LocationType and Location records.

  Background:
    Given I have appropriate Nautobot permissions:
      | model             | permissions            |
      | dcim.locationtype | add,change,delete,view |
      | dcim.location     | add,change,delete,view |
    And I am logged in to Nautobot

  Scenario: Creating a hierarchy of LocationTypes
    When I navigate to the "Add LocationType" view
    And I make the following form entries:
      | name              | nestable |
      | Geographic Region | True     |
    And I submit the form
    Then a LocationType exists in Nautobot with the following properties:
      | name              | parent | nestable |
      | Geographic Region | None   | True     | 
    And Nautobot shows the "LocationType Detail" view for LocationType with name "Geographic Region"
    When I click the "Add child" button
    Then Nautobot shows the "Add LocationType" view with the following presets:
      | parent            |
      | Geographic Region |
    When I make the following form entries:
      | name | 
      | city |
    And I submit the form
    Then a LocationType exists in Nautobot with the following properties:
      | name | parent              | nestable |
      | City | Geographic Region   | False    | 
    And Nautobot shows the "LocationType Detail" view for LocationType with name "City"
    When I click the "Add child" button
    Then Nautobot shows the "Add LocationType" view with the following presets:
      | parent |
      | City   |
    When I make the following form entries:
      | name     | content_types |
      | Building | [dcim.device] |
    And I submit the form
    Then a LocationType exists in Nautobot with the following properties:
      | name     | parent | nestable | content_types |
      | Building | City   | False    | [dcim.device] |
    And Nautobot shows the "LocationType Detail" view for LocationType with name "Building"

  Scenario: Creating a hierarchy of Locations
    Given that LocationType records exist with the following properties:
      | name              | parent            | nestable |
      | Geographic Region | None              | True     |
      | City              | Geographic Region | False    |
      | Building          | Building          | False    |
    When I navigate to the "Add Location" view
    And I make the following form entries:
      | name          | location_type     | status |
      | United States | Geographic Region | Active |
    And I submit the form
    Then a Location exists in Nautobot with the following properties:
      | name          | location_type     | parent | status |
      | United States | Geographic Region | None   | Active |
    And Nautobot shows the "Location Detail" view for Location with name "United States"

    When I click the "Add child" button
    Then Nautobot shows the "Add Location" view with the following presets:
      | parent        |
      | United States |
    When I make the following form entries:
      | name           | location_type     | status | time_zone |
      | North Carolina | Geographic Region | Active | America/New York |
    And I submit the form
    Then a Location exists in Nautobot with the following properties:
      | name           | location_type     | parent        | status | time_zone        |
      | North Carolina | Geographic Region | United States | Active | America/New York |
    And Nautobot shows the "Location Detail" view for Location with name "North Carolina"

    When I click the "Add child" button
    Then Nautobot shows the "Add Location" view with the following presets:
      | parent         |
      | North Carolina |
    When I make the following form entries:
      | name   | location_type | status  |
      | Durham | City          | Planned |
    And I submit the form
    Then a Location exists in Nautobot with the following properties:
      | name   | location_type | parent         | status  |
      | Durham | City          | North Carolina | Planned |
    And Nautobot shows the "Location Detail" view for Location with name "Durham"

    When I click the "Add child" button
    Then Nautobot shows the "Add Location" view with the following presets:
      | parent |
      | Durham |
    When I make the following form entries:
      | name                       | location_type | status  | physical_address     |
      | Durham Bulls Athletic Park | Building      | Planned | 409 Blackwell Street |
    And I submit the form via the "Create and Add Another" button
    Then a Location exists in Nautobot with the following properties:
      | name                       | location_type | parent | status  | physical_address     |
      | Durham Bulls Athletic Park | Building      | Durham | Planned | 409 Blackwell Street |
    And Nautobot shows the "Add Location" view with the following presets:
      | parent | location_type | status  |
      | Durham | Building      | Planned |
    When I make the following form entries:
      | name                          |
      | Durham Performing Arts Center |
    And I submit the form
    Then a Location exists in Nautobot with the following properties:
      | name                          | location_type | parent | status  |
      | Durham Performing Arts Center | Building      | Durham | Planned |
    And Nautobot shows the "Location Detail" view for Location with name "Durham Performing Arts Center"

    When I click on the "Durham" link
    Then Nautobot shows the "Location Detail" view for Location with name "Durham"
    And the table "Children" includes rows with the following properties:
      | name                          | status  |
      | Durham Bulls Athletic Park    | Planned |
      | Durham Performing Arts Center | Planned |

  Scenario Outline: Missing required fields for Location
    Given that LocationType records exist with the following properties:
      | name              | parent            | nestable |
      | Geographic Region | None              | True     |
      | City              | Geographic Region | False    |
      | Building          | Building          | False    |
    When I navigate to the "Add Location" view
    And I make the following form entries:
      | name   | location_type   | parent   | status   |
      | <name> | <location_type> | <parent> | <status> |
    And I submit the form
    Then Nautobot shows the "Add Location" view with the following presets:
      | name   | location_type   | parent   | status   |
      | <name> | <location_type> | <parent> | <status> |
    And the field "<required_field>" shows error message "<error_message>"

    Scenarios:
      | name          | location_type     | parent | status | required_field | error_message                      |
      |               | Geographic Region | None   | Active | name           | Please fill out this field.        |
      | United States | None              | None   | Active | location_type  | Please select an item in the list. |
      | United States | Geographic Region | None   | None   | status         | Please select an item in the list. |
      | Durham        | City              | None   | Active | parent         | A Location of type City must have a parent Location. |
```

**Comment by gsnider2195** [comment]
```
    And I submit the form
    Then a LocationType exists in Nautobot with the following properties:
      | name              | parent | nestable |
      | Geographic Region | None   | True     | 
    And Nautobot shows the "LocationType Detail" view for LocationType with name "Geographic Region"
```

Perhaps change that last line to `And Nautobot redirects to the "LocationType Detail" view for LocationType with name "Geographic Region"`

**Comment by glennmatthews** [comment]
Implemented by #4803
